### ISSUE TITLE
Add col types to read_movement_data

### ISSUE THREAD

**Issue by CarlijnB** [digivet-consortium/movenet#2]
- [ ] Check date format reading for fread and read_delim
- [ ] Incorporate fixed column types (char for IDs, int for nr_pigs, date / datetime for dates)

**Comment by CarlijnB** [comment]
@mdenwood For maximum flexibility on date formats, what do you think of initially reading in dates /datetimes as strings, and passing them through lubridate::guess_formats to turn them into dates? 
Or have users provide the correct date format as an option?
(To me either seems preferable to directly reading in dates, where this depends on some locale or standardized format)

**Comment by mdenwood** [comment]
Yes I think we will need to parse the date/time column as a string and then use some kind of parsing in R depending on the expected format.  My preference would be for the default to use readr::parse_datetime rather than lubridate::guess_formats i.e. assume that the date/time is ISO-8601 compliant unless the user specifies an explicit format string.  But that can be adjusted once we know what the date/time formats will look like for each of the data files.

**Comment by CarlijnB** [comment]
Hi @mdenwood, thanks for your comments.
I've now replaced fread with readr::read_delim (though without understanding how that would be preferable).
I've allowed for optional user-defined delimiter and date format.
Due to ordering of various actions within read_delim, I've had to use a few separate functions now to rename columns and set column type, so I'm importing a bit more
